File: python/widest-vertical-area-between-two-points-containing-no-points.py
class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        points.sort(key=lambda point: point[0])
        diff = 0
        for i in range(1, len(points)):
            diff = max(points[i][0] - points[i - 1][0], diff)
        return diff

File: go/widest-vertical-area-between-two-points-containing-no-points.go
import (
    "sort"
)

func maxWidthOfVerticalArea(points [][]int) int {
    sort.Slice(points, func (i int, j int) bool {
        return points[i][0] < points[j][0]
    })
    maxDiff := 0
    for i := 1; i < len(points); i++ {
        diff := points[i][0] - points[i - 1][0]
        if maxDiff < diff {
            maxDiff = diff
        }
    }
    return maxDiff
}
